ixr7220h4-64d/sonic_platform/eeprom.py

```diff
@@ -0,0 +1,175 @@
+"""
+    Nokia 7220 IXR H4-64D
+
+    Module contains platform specific implementation of SONiC Platform
+    Base API and provides the EEPROMs' information.
+
+    The different EEPROMs available are as follows:
+    - System EEPROM : Contains Serial number, Service tag, Base MA
+                      address, etc. in ONIE TlvInfo EEPROM format.
+"""
+
+try:
+    import os
+    from sonic_platform_base.sonic_eeprom.eeprom_tlvinfo import TlvInfoDecoder
+    from sonic_py_common import logger
+except ImportError as e:
+    raise ImportError(str(e) + ' - required module not found') from e
+
+sonic_logger = logger.Logger('eeprom')
+
+class Eeprom(TlvInfoDecoder):
+    """Nokia platform-specific EEPROM class"""
+
+    I2C_DIR = "/sys/bus/i2c/devices/"
+    FAN_EEPROM_DIR = "/sys/bus/i2c/devices/26-0051/eeprom"
+    def __init__(self, is_psu, psu_index, is_fan, drawer_index):
+        self.is_psu_eeprom = is_psu
+        self.is_fan_eeprom = is_fan
+        self.is_sys_eeprom = not is_psu | is_fan
+        self.service_tag = 'NA'
+        self.part_number = 'NA'
+
+        if self.is_sys_eeprom:
+            self.start_offset = 0
+            self.eeprom_path = self.I2C_DIR + "20-0051/eeprom"
+            # System EEPROM is in ONIE TlvInfo EEPROM format
+            super().__init__(self.eeprom_path, self.start_offset, '', True)            
+            self.base_mac = os.popen("sudo decode-syseeprom -d -m").read().rstrip('\n')
```

Comment:
To use os.popen will cause the below warning under unit test.

```
admin@logan:~$ time sudo python3 /usr/lib/python3/dist-packages/sonic_platform/test/test-fan-drawer.py
---------------------------
Chassis Fan Drawer Unit Test
---------------------------
/usr/lib/python3.11/subprocess.py:1125: ResourceWarning: subprocess 3988185 is still running
  _warn("subprocess %s is still running" % self.pid,
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3/dist-packages/sonic_platform/eeprom.py:51: ResourceWarning: unclosed file <_io.TextIOWrapper name=4 encoding='UTF-8'>
  self.base_mac = os.popen("ip link show eth0 | grep ether | awk '{print $2}'").read().rstrip('\n')
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/usr/lib/python3.11/subprocess.py:1125: ResourceWarning: subprocess 3988189 is still running
  _warn("subprocess %s is still running" % self.pid,
ResourceWarning: Enable tracemalloc to get the object allocation traceback
```

Advise to use subprocess.Popen shown below:

     out, _ = self.run_command(["sudo", "decode-syseeprom", "-d"])
    def run_command(self, command):
        """
        Run bash command.
        """
        with subprocess.Popen(command, shell=False, text=True, stdout=subprocess.PIPE) as proc:
            output = proc.communicate()[0]
            return output, proc.returncode